src/frontend/src/icp/components/transactions/IcTransaction.svelte

```diff
@@ -36,9 +37,13 @@
 
 	let icon: ComponentType;
 	$: icon = ['burn', 'approve'].includes(transactionType)
-		? IconBurn
+		? pending
```

Comment:
Instead of duplicating `pending` shouldn't it be the first criteria?

```
$: icon = pending ? IconConvert : same_as_on_main...
```

Reply:
yes and no, since it should be only for `burn`, `approve` and `mint` (and not `send` and `receive`). So it would be something like

```typescript
	$: icon =
		['burn', 'approve', 'mint'].includes(transactionType) && pending
			? IconConvert
			: ['burn', 'approve'].includes(transactionType)
				? IconBurn
				: transactionType === 'mint'
					? IconMint
					: incoming === false
						? IconSend
						: IconReceive;
```

WDYT?

Reply:
Sounds good.